internal/buff/write.go

```diff
@@ -0,0 +1,167 @@
+// This source file is part of the EdgeDB open source project.
+//
+// Copyright 2020-present EdgeDB Inc. and the EdgeDB authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buff
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/edgedb/edgedb-go/types"
+)
+
+// Writer is a write buffer.
+type Writer struct {
+	alocatedMemory [256 * 1024]byte
```

Comment:
This is a very big buffer. Our WriteBuffer in Python starts with just 1kb, which is actually plenty for most cases. It can grow on demand in chunk increments. We need to implement the same strategy here.

Reply:
Ok. I was making up numbers. I'll change that. It already grows if needed :+1:

Reply:
done :heavy_check_mark: